File: server/routes/passwordRoute.js
const express = require('express');
const router = express.Router(); // ✅ Define router BEFORE using it
const Admin = require('../models/Admin');
const Streamer = require('../models/Streamer');

// POST /api/password/verify-password
router.post('/verify-password', async (req, res) => {
  const { username, userType, currentPassword } = req.body;

  try {
    const Model = userType === 'admin' ? Admin : Streamer;
    const user = await Model.findOne({ username });

    if (!user) {
      return res.status(404).json({ error: 'User not found.' });
    }

    if (currentPassword !== user.password) {
      return res.status(401).json({ error: 'Current password is incorrect.' });
    }

    res.status(200).json({ message: 'Password verified.' });
  } catch (err) {
    console.error('Password verification error:', err);
    res.status(500).json({ error: 'Server error.' });
  }
});

// POST /api/password/update-password
router.post('/update-password', async (req, res) => {
  const { username, userType, newPassword } = req.body;

  try {
    const Model = userType === 'admin' ? Admin : Streamer;
    const user = await Model.findOne({ username });

    if (!user) {
      return res.status(404).json({ error: 'User not found.' });
    }

    user.password = newPassword;
    await user.save();

    res.status(200).json({ message: 'Password updated successfully.' });
  } catch (err) {
    console.error('Password update error:', err);
    res.status(500).json({ error: 'Server error.' });
  }
});

module.exports = router;

File: backend/server/routes/passwordRoute.py
from fastapi import APIRouter, HTTPException, Request
from pydantic import BaseModel

router = APIRouter()

class VerifyPasswordRequest(BaseModel):
    username: str
    userType: str
    currentPassword: str

class UpdatePasswordRequest(BaseModel):
    username: str
    userType: str
    newPassword: str

@router.post("/verify-password")
def verify_password(data: VerifyPasswordRequest, request: Request):
    db = request.app.state.user_db
    collection = db["admin"] if data.userType.lower() == "admin" else db["streamer"]

    user = collection.find_one({"username": data.username})
    if not user:
        raise HTTPException(status_code=404, detail="User not found")

    if user["password"] != data.currentPassword:
        raise HTTPException(status_code=401, detail="Current password is incorrect")

    return {"message": "Password verified."}

@router.post("/update-password")
def update_password(data: UpdatePasswordRequest, request: Request):
    db = request.app.state.user_db
    collection = db["admin"] if data.userType.lower() == "admin" else db["streamer"]

    user = collection.find_one({"username": data.username})
    if not user:
        raise HTTPException(status_code=404, detail="User not found")

    collection.update_one(
        {"username": data.username},
        {"$set": {"password": data.newPassword}}
    )

    return {"message": "Password updated successfully."}